march_hardware/src/PDOmap.cpp

```diff
@@ -124,7 +126,7 @@ std::map<IMCObjectName, int> PDOmap::configurePDO(int slaveIndex, int baseRegist
   currentReg++;
   if (currentReg <= (baseRegister + nr_of_regs))
   {
-    for (int unusedRegister = currentReg; unusedRegister < baseRegister + this->nr_of_regs; unusedRegister++)
+    for (int unusedRegister = currentReg; unusedRegister <= (baseRegister + this->nr_of_regs); unusedRegister++)
```

Comment:
I think the if and the for loop conditions should both be less than (not equal). Now you will write sdo messages for unused registers up to 0x1604, while the PDO registers only go up till 0x1603.